src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +51,12 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .mapToObj(index -> index % 2 != 0 ? numbers.get(index) - 1 : numbers.get(index))
+                .filter(index -> index % 2 != 0)
+                .mapToDouble(Double::valueOf)
```

Comment:
Let's use 1 map operation instead of two, and we should be careful with Double values comparison as != may cause bugs so let's use a safer way to compare.
```suggestion
                .mapToDouble(index -> index % 2 != 0 ? numbers.get(index) - 1 : numbers.get(index))
                .filter(index -> index % 2 != 0)
```